src/operators.jl

```diff
@@ -251,7 +251,7 @@ _sizehint_expr!(q, n) = nothing
 #############################################################################
 
 # TODO: specialize sum for DenseAxisArray and SparseAxisArray of JuMP objects?
-Base.sum(vars::Array{<:AbstractVariableRef}) = GenericAffExpr(0.0, [v => 1.0 for v in vars])
+Base.sum(vars::AbstractArray{<:AbstractVariableRef}) = GenericAffExpr(0.0, [v => 1.0 for v in vars])
 function Base.sum(array::AbstractArray{<:AbstractVariableRef})
```

Comment:
Now this one can be removed

Reply:
Removing this sum method is failing because creating this kind of array `[v => 1.0 for v in x]` 
in a `JuMPDict{VariableRef}`
```julia
m = Model()
@variable(m, x[1:3])
[v => 1.0 for v in x]
S = [1, 3]
@variable(m, y[S])
[v => 1.0 for v in y] #cannot create the array
[v => 1.0 for v in y.data] #creates what we want
```
Maybe we could leave this method and add more specialized ones, something like 
```julia
Base.sum(vars::Array{<:AbstractVariableRef}) = GenericAffExpr(0.0, [v => 1.0 for v in vars]) 
Base.sum(vars::Containers.DenseAxisArray{<:AbstractVariableRef}) = GenericAffExpr(0.0, [v => 1.0 for v in vars.data])
function Base.sum(array::AbstractArray{<:AbstractVariableRef}) 
     result_expression = zero(eltype(array)) 
     for variable in array 
         add_to_expression!(result_expression, variable) 
     end 
     return result_expression 
 end 
```

Reply:
The other option would be 
```julia
function Base.sum(array::AbstractArray{<:AbstractVariableRef}) 
     result_expression = zero(eltype(array)) 
     for variable in array 
         add_to_expression!(result_expression, variable) 
     end 
     return result_expression 
 end 
```
and rely that the `add_to_expression!` function is well specialized.